File: Problem9.py

def pythagorean_theorem(a, b, c):
    return a**2 + b**2 == c**2

def main(limit):
    for a in range(1, limit):
        for b in range(1, limit):
            for c in range(1, limit):
                if a + b + c == limit:
                    if pythagorean_theorem(a, b, c):
                        print(a, b, c)
                        print(a * b * c)
                        return


if __name__ == '__main__':
    main(1000)

File: Problem9.rs

fn pythagorean_theorem(a: i32, b: i32, c: i32) -> bool {
    return a.pow(2) + b.pow(2) == c.pow(2);
}

fn main() {
    let limit = 1000;

    for a in 1..limit {
        for b in 1..limit {
            for c in 1..limit {
                if a + b + c == limit {
                    if pythagorean_theorem(a, b, c) {
                        println!("{} {} {}", a, b, c);
                        println!("{}", a * b * c);
                    }
                } 
            } 
        } 
    }
}
